src/main/java/com/microsoft/sqlserver/jdbc/SQLServerXAConnection.java

```diff
@@ -101,21 +104,32 @@ public final class SQLServerXAConnection extends SQLServerPooledConnection imple
     }
 
     @Override
-    public synchronized XAResource getXAResource() throws java.sql.SQLException {
+    public XAResource getXAResource() throws java.sql.SQLException {
         // All connections handed out from this physical connection have a common XAResource
         // for transaction control. IE the XAResource is one to one with the physical connection.
-
-        if (XAResource == null)
-            XAResource = new SQLServerXAResource(getPhysicalConnection(), physicalControlConnection, toString());
-        return XAResource;
+        SQLServerXAResource result = XAResource;
```

Comment:
Similar to above, why can't this just be wrapped in a try / catch block like the other changes.

Reply:
Refer to https://en.wikipedia.org/wiki/Double-checked_locking

"Note the local variable "localRef", which seems unnecessary. The effect of this is that in cases where helper is already initialized (i.e., most of the time), the volatile field is only accessed once (due to "return localRef;" instead of "return helper;"), which can improve the method's overall performance by as much as 40 percent."

One could use the VarHandles stuff for even better performance but that's even trickier.